argoverse/evaluation/eval_detection.py

```diff
@@ -0,0 +1,329 @@
+# <Copyright 2020, Argo AI, LLC. Released under the MIT license.>
+import argparse
+import logging
+import os
+from collections import defaultdict
+from dataclasses import Field, dataclass, field
+from multiprocessing import Pool
+from pathlib import Path
+from typing import DefaultDict, List, Tuple
+
+import matplotlib
+import numpy as np
+import pandas as pd
+from pandas.core import frame
+
+from argoverse.data_loading.object_classes import OBJ_CLASS_MAPPING_DICT
+from argoverse.data_loading.object_label_record import read_label
+from argoverse.evaluation.detection_utils import (
+    DistFnType,
+    FilterMetric,
+    SimFnType,
+    compute_affinity_matrix,
+    dist_fn,
+    filter_instances,
+    get_ranks,
+    interp,
+)
+
+matplotlib.use("Agg")  # isort:skip
+
+
+import matplotlib.pyplot as plt  # isort:skip
+
+
+logger = logging.getLogger(__name__)
+
+
+TP_METRIC_NAMES = ["ATE", "ASE", "AOE"]
+NUM_TP_METRICS = len(TP_METRIC_NAMES)
+
+METRIC_NAMES: List[str] = ["AP"] + TP_METRIC_NAMES + ["CDS"]
+METRIC_DEFAULT_VALUES: List[float] = [0.0, 1.0, 1.0, 1.0, 0.0]
+
+
+@dataclass
+class DetectionCfg:
+    """Instantiates a DetectionCfg object for configuring a DetectionEvaluator.
+
+    Args:
+        affinity_threshs: The affinity thresholds for determining a true positive.
+        affinity_fn_type: The type of affinity function to be used for calculating average precision.
+        n_rec_samples: Number of recall points to sample uniformly in [0, 1]. Default to 101 recall samples.
+        tp_thresh: Center distance threshold for the true positive metrics (in meters).
+        significant_digits: The precision for metrics.
+        detection_classes: Detection classes for evaluation.
+        detection_metric: The detection metric to use for filtering of both detections and ground truth annotations.
+        max_detection_range: The max distance (under a specific metric) for a a detection or ground truth to be considered for evaluation.
+        save_figs: Flag to save figures.
+    """
+
+    affinity_threshs: List[float] = field(default_factory=lambda: [0.5, 1.0, 2.0, 4.0])
+    affinity_fn_type: SimFnType = SimFnType.CENTER
+    n_rec_samples: int = 101
+    tp_thresh: float = 2.0
+    significant_digits: int = 3
+    detection_classes: List[str] = field(default_factory=lambda: list(OBJ_CLASS_MAPPING_DICT.keys()))
+    detection_metric: FilterMetric = FilterMetric.EUCLIDEAN
+    max_detection_range: float = 50.0
+    save_figs: bool = False
+    tp_normalization_terms: np.ndarray = field(init=False)
+
+    def __post_init__(self):
+        self.tp_normalization_terms: np.ndarray = np.array([self.tp_thresh, 1.0, np.pi])
```

Comment:
`np.pi` should be a module-level `MAX_YAW_ERROR = np.pi`, instead of a magic number here, in my opinion